### ISSUE TITLE
Enqueued messages in ReliableQueue dequeue as collection of zeros or null values.

### ISSUE THREAD

**Issue by wmeddie** [Azure/service-fabric-issues#19]
After an application upgrade my reliable services dequeue zero-filled values.

I suspected a Serialization problem, but I haven't touched the class definition in over 2 months and the queue data is less than an hour old.

 The problems started sometime around the 5.1.150.9590 upgrade.

I have a Task class defined as

```
[DataContract]
internal class FetchTask
{
    public Guid Guid { get; set; }
    public string ResourceUrl { get; set; }
    public int Tries { get; set; }

    public FetchTask(string url)
    {
        this.Guid = Guid.NewGuid();
        this.ResourceUrl = url;
        this.Tries = 0;
    }
}
```

which I enqueue with: 

```
var jobQueue = await StateManager.GetOrAddAsync<IReliableQueue<FetchTask>>(JobQueue);

using (var tx = StateManager.CreateTransaction())
{
    var task = new FetchTask(url);
    await jobQueue.EnqueueAsync(tx, task);
    await tx.CommitAsync();
}
```

Which fails inside RunAsync due to url being null.  While a null URL could be from the upstream caller I see the following error log in my traces:

`Error: Processing job {00000000-0000-0000-0000-000000000000}`

Which is suspicious because it's highly unlikely Guid.NewGuid() would return the 0 GUID and impossible that it would return it thousands of times.

I suspect not having `[DataMember]` is the cause but I wonder why I was able to upgrade the application just fine (about 40 times) before 5.1.150.9590.

**Comment by wmeddie** [comment]
Traces also show that I'm not inserting the 0 GUID into the queue either:

`End: Enqueueing job {22103c4a-02cf-4e05-8326-ea28578557cd}`

**Comment by vturecek** [comment]
Thanks for reporting this. We were able to repro it too and found a bug where values are not paged into memory from disk correctly, resulting in default values (data is still stored correctly on disk though). We will be rolling out a fix for this ASAP.

**Comment by vturecek** [comment]
This issue was fixed as of 5.1.163.
